src/verifai/falsifier.py

```diff
@@ -98,40 +116,73 @@ def analyze_error_table(self, analysis_params= None, error=None):
                 self.safe_analysis.random_samples = \
                     [self.samples[i] for i in self.safe_analysis.random]
 
+    def get_confidence_interval(self, confidence_level=0.95):
+        N = len(self.samples)
+        c = len(self.error_table.table)
+        return proportion_confint(c, N, alpha=1 - confidence_level, method='beta')
+
     def run_falsifier(self):
         i = 0
         ce_num = 0
+        counterexamples = []
+        server_samples = []
+        rhos = []
+        self.total_sample_time = 0
+        self.total_simulate_time = 0
+        # print(f'Running falsifier; server class is {type(self.server)}')
+        if self.n_iters is not None:
+            bar = progressbar.ProgressBar(max_value=self.n_iters)
+        else:
+            widgets = ['Scenes generated: ', progressbar.Counter('%(value)d'),
+               ' (', progressbar.Timer(), ')']
+            bar = progressbar.ProgressBar(widgets=widgets)
+        # bar = progressbar.ProgressBar(max_value=self.n_iters)
         while True:
-            if i == self.n_iters:
-                break
             try:
-                sample, rho = self.server.run_server()
+                sample, rho, (sample_time, simulate_time) = self.server.run_server()
+                self.total_sample_time += sample_time
+                self.total_simulate_time += simulate_time
+                # print(f'sample = {sample}, rho = {rho}')
             except TerminationException:
                 if self.verbosity >= 1:
                     print("Sampler has generated all possible samples")
                 break
-            if self.verbosity >= 1:
-                print("Sample no: ", i, "\nSample: ", sample, "\nRho: ", rho)
+            # if self.verbosity >= 1:
+            #     print("Sample no: ", i, "\nSample: ", sample, "\nRho: ", rho)
             self.samples[i] = sample
-            if isinstance(rho, (list, tuple)):
-                check_var = rho[-1]
```

Comment:
Yes - whether or not the rho value corresponds to a counterexample needs to be directly handled by the sampler now, since rho being a list or tuple could also mean that the monitor is a multi_objective_monitor.